docs/dev/hacking-guide.md

```diff
@@ -0,0 +1,177 @@
+
+```bash
+oc edit deployment/cluster-cloud-controller-manager -n openshift-cloud-controller-manager-operator
+```
+
+After changing the download reference in the images ConfigMap an easy way to swap out the controller is to let the CCCMO do it for you. You can delete the deployment associated with the cloud provider CCM and then the CCCMO will create a new one for you, like this:
+
+```bash
+oc delete deployment/aws-cloud-controller-manager -n openshift-cloud-controller-manager
+```
+
+## How to build operator in combination with other operators and build a release image
+
+At the current stage of development it is frequent that for testing a feature, another openshift operator change should be in place to make the feature work in CCCMO.
+
+To achieve this you need to build custom release payload. To do that you should use `oc adm release new`.
+
+Here is an example how to build a release with changes in CCCMO, KCMO and MCO at the same time. Precondition for this step - you have to build each operator image individually and push to a public registry.
+
+```bash
+oc adm release new --server=https://api.ci.l2s4.p1.openshiftapps.com:6443 \ # CI server to pull existing release payload from
+  --from-release=registry.ci.openshift.org/ocp/release:4.8.0-0.ci-2021-05-24-100659 \ # Your release tag to pull the rest of the images from
+  --registry-config=~/pull-secret \ # Your pull-secret with permissions to read from CI server
+  --to-image=quay.io/dgrigore/release:latest \ # Where the release image will be pushed
+  # Now your image overrides for image tags
+    machine-config-operator=quay.io/dgrigore/machine-config-operator:ccm \
+    cluster-kube-controller-manager-operator=quay.io/dgrigore/cluster-kube-controller-manager-operator:azure-bootstrap \
+    cluster-cloud-controller-manager-operator=quay.io/dgrigore/cluster-cloud-controller-manager-operator:azure-bootstrap
+```
+
+At the current stage of development operator is only present in CI builds, but is not present in nightlies and is not in release. You could pick one of available CI releases from the list here: https://amd64.ocp.releases.ci.openshift.org/#4.8.0-0.ci.
+You might have to pick a fresh one once in couple of days, as they are constatntly being removed, and run `oc adm release new` once again.
+
+Tags for image overrides are coming from the operator manifests itself. The image tags could be found in `manifests/image-references`. Example might be KCMO image references: https://github.com/openshift/cluster-kube-controller-manager-operator/blob/master/manifests/image-references
+In this scenario, you build KCMO operator image with `make image`, push it somewhere like `quay.io/dgrigore/cluster-kube-controller-manager-operator:azure-bootstrap` and then specify the override like: 
+
+```bash
+...
+cluster-kube-controller-manager-operator=quay.io/dgrigore/cluster-kube-controller-manager-operator:azure-bootstrap 
+...
+```
```

Comment:
Might be worth making these examples more generic, so they don't directly relate to your repo
```suggestion
In this scenario, you build KCMO operator image with `make image`, push it somewhere like `quay.io/<my-repo>/cluster-kube-controller-manager-operator:<my-branch>` and then specify the override like: 

```bash
...
cluster-kube-controller-manager-operator=quay.io/<my-repo>/cluster-kube-controller-manager-operator:<my-branch>
...
```
```

Reply:
++ i was thinking something similar